test/integration/util/util.go

```diff
@@ -127,7 +132,10 @@ func StartFakePVController(ctx context.Context, clientSet clientset.Interface, i
 			claimRef := obj.Spec.ClaimRef
 			pvc, err := clientSet.CoreV1().PersistentVolumeClaims(claimRef.Namespace).Get(ctx, claimRef.Name, metav1.GetOptions{})
 			if err != nil {
```

Comment:
This should avoid any races:

```suggestion
			if err != nil && errors.Is(err, context.Canceled) {
```

Reply:
We have to return when there was an error, otherwise the code below would use a nil `pvc`. What needs to be fixed is the check whether that error should be logger. I had the logic backwards. It now is:

```
			if err != nil {
				// Note that the error can be anything, because components like
				// apiserver are also shutting down at the same time, but this
				// check is conservative and only ignores the "context canceled"
				// error while shutting down.
				if ctx.Err() == nil || !errors.Is(err, context.Canceled) {
					klog.Errorf("error while getting %v/%v: %v", claimRef.Namespace, claimRef.Name, err)
				}
				return
			}
```

Reply:
sg